shiksha-platform/frontend-modulefederation
# Refactored the QuestionBox Component in Worksheet

## Description
Refactored the **QuestionBox** component in the module **Worksheet** as part of the refactoring progress

### Changes
* added interfaces wherever applicable
* created a new component HTMLPrint, that renders the given HTML in a div and returns it, shifted it from outside index.tsx to make it easier to use in other components(if required)
* extracted one component 'QuestionOption' from QuestionBox to increase readibility

## How to test
1. Run the module using the command: ```lerna run start --parallel --scope='{worksheet,@shiksha/common-lib,core,students,attendance}'```
2. The module should work fine as no other changes have been added, other than refactoring and enhancing modularity.

## Refactored Components
### QuestionOption
![feature](https://user-images.githubusercontent.com/82583557/183290190-fdb886b1-2005-4718-af59-85548eb962fa.gif)

### HTMLPrint
![feature gif](https://user-images.githubusercontent.com/82583557/183290247-c934e80b-776c-4017-a33f-b67affe47340.gif)
